msaad.go

```diff
@@ -31,15 +31,17 @@ import (
 
 // ConfigMSAAD is the JSON definition for the Microsoft Azure Active Directory synchronization settings
 type ConfigMSAAD struct {
-	Verbose              bool   // If true, then emit verbose logging
-	DryRun               bool   // If true, don't actually take any action, just log the intended actions
-	TenantID             string // Your tenant UUID (ie ID of your AAD instance)
-	ClientID             string // Your client UUID (ie ID of your application)
-	ClientSecret         string
-	MergeIntervalSeconds int               // If non-zero, then overrides the merge interval
-	EssentialRoles       []string          // List of principleName of AAD roles. If this list is not empty, then the user must belong to at least one of these groups, in order to be created in the Authaus DB
-	RoleToGroup          map[string]string // Map from principleName of AAD role, to Authaus group.
-	AllowArchiveUser     bool              // If true, then archive users who no longer have the relevant roles in the AAD
+	Verbose                 bool              // If true, then emit verbose logging
+	DryRun                  bool              // If true, don't actually take any action, just log the intended actions
+	TenantID                string            // Your tenant UUID (ie ID of your AAD instance)
+	ClientID                string            // Your client UUID (ie ID of your application)
+	ClientSecret            string            // Secrets used for authenticating Azure AD requests
+	MergeIntervalSeconds    int               // If non-zero, then overrides the merge interval
+	DefaultRoles            []string          // Roles that are activated by default if a user has any one of the AAD roles
```

Comment:
Doesn't this break the conceptual pattern that having a DefaultRole is supposed supposed to solve? I was under the impression that the client wanted EssentialRoles to be removed from the integration.

We'll just need to chat, because from where I am standing, I can't see a world in which a permission like "isActive" can be both essential and default at the same time. How would they need it to remain active, but then be assigned it if they have any of the other permissions, and remain active?

OR is this an option that you want to live in parallel with this DefaultRoles and have turned on by config?

Reply:
Parallel - yes 👍

Reply:
But, don't worry about that now.